node/cmd/ccq/utils.go

```diff
@@ -260,19 +272,24 @@ func validateRequest(logger *zap.Logger, env common.Environment, perms Permissio
 			}
 		case *query.EthCallByTimestampQueryRequest:
 			for _, callData := range q.CallData {
-				if q.TargetTimestamp == 0 {
-					logger.Debug("eth call by timestamp must have a non-zero timestamp", zap.String("userName", permsForUser.userName))
-					return http.StatusBadRequest, fmt.Errorf("eth call by timestamp must have a non-zero timestamp")
+				contractAddress, err := vaa.BytesToAddress(callData.To)
+				if err != nil {
+					logger.Debug("failed to parse contract address", zap.String("userName", permsForUser.userName), zap.String("contract", hex.EncodeToString(callData.To)), zap.Error(err))
+					return http.StatusBadRequest, fmt.Errorf("failed to parse contract address: %w", err)
 				}
-				// TODO: For now, the block hints are required!
-				if q.TargetBlockIdHint == "" {
-					logger.Debug("eth call by timestamp must have the target block hint", zap.String("userName", permsForUser.userName))
-					return http.StatusBadRequest, fmt.Errorf("eth call by timestamp must have the target block hint")
+				if len(callData.Data) < ETH_CALL_SIG_LENGTH {
```

Comment:
Do you need to check if it is too long?

Reply:
No, it can be any arbitrary length, based on the query they are doing.